src/expr/src/scalar/func.rs

```diff
@@ -5752,10 +5751,13 @@ impl VariadicFunc {
             }
             ArrayToString { .. } => ScalarType::String.nullable(true),
             ListCreate { elem_type } => {
-                soft_assert!(
-                    input_types.iter().all(|t| t.scalar_type.base_eq(elem_type)),
-                    "Args to ListCreate should have types that are compatible with the elem_type"
-                );
+                // commented out to work around
+                // https://github.com/MaterializeInc/materialize/pull/9008
```

Comment:
Please don't reference the pull request but the issue

Reply:
fixed